components/infobox/wikis/halo/infobox_map_custom.lua

```diff
@@ -0,0 +1,91 @@
+---
+-- @Liquipedia
+-- wiki=halo
+-- page=Module:Infobox/Map
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Map = require('Module:Infobox/Map')
+local Injector = require('Module:Infobox/Widget/Injector')
+local Cell = require('Module:Infobox/Widget/Cell')
+local MapModes = require('Module:MapModes')
+local String = require('Module:String')
+
+local CustomMap = Class.new()
+
+local CustomInjector = Class.new(Injector)
+
+local _args
+local _game
+
+local _GAME = mw.loadData('Module:GameVersion')
+
+function CustomMap.run(frame)
+	local customMap = Map(frame)
+	customMap.createWidgetInjector = CustomMap.createWidgetInjector
+	customMap.getCategories = CustomMap.getCategories
+	_args = customMap.args
+	return customMap:createInfobox(frame)
+end
+
+function CustomMap:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomInjector:addCustomCells(widgets)
+	table.insert(widgets, Cell{
+		name = 'Location',
+		content = {_args.location}
+	})
+	table.insert(widgets, Cell{
+		name = 'Type',
+		content = {_args.type}
+	})
+	table.insert(widgets, Cell{
+		name = 'Max Players',
+		content = {_args.players}
+	})
+	table.insert(widgets, Cell{
+		name = 'Game Version',
+		content = {CustomMap._getGameVersion()},
+		options = {makeLink = true}
+	})
+	table.insert(widgets, Cell{
+		name = 'Game Modes',
+		content = {CustomMap._getGameMode()}
+	})
+	return widgets
+end
+
+function CustomMap._getGameVersion()
+	local game = string.lower(_args.game or '')
+	_game = _GAME[game]
+	return _game
+end
+
+function CustomMap._getGameMode()
+
+	if String.isEmpty(_args.mode) then
+		return nil
+	end
+	local modeIcon = MapModes.get({mode = _args.mode, date = _args.releasedate, size = 15})
+	local mapModeDisplay = modeIcon .. ' [[' .. _args.mode .. ']]'
+	return mapModeDisplay
+end
+
+function CustomMap:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomMap:addToLpdb(lpdbData)
+	lpdbData.extradata.creator2 = mw.ext.TeamLiquidIntegration.resolve_redirect(_args.creator2)
+	lpdbData.extradata.type = _args.type
+	lpdbData.extradata.players = _args.players
+	lpdbData.extradata.game = _game
+	lpdbData.extradata.modes = _args.mode
```

Comment:
".modes" does that indicate that maps can have several modes
if so the "._getGameMode" function should account for that

Reply:
._getGameMode should account for multiple modes

Reply:
okay, so how should the input look like?
`|mode1=A|mode2=B|...` or `|mode=A,B,C` (or some other separator then `,`)?

depending on that you need to adjust stuff because the current implementation of `_getGameMode` does not.

Reply:
``|mode=|mode1=|mode2=.......`` This way makes the most sense considering the game mode flag is being added in-front of the modes link. Otherwise the ``,`` would be used. I'm currently testing the use of ``getAllArgsForBase`` but I'm starting to think this isn't the implementation I want

Reply:
well if you got a `,` separated list you could just use `mw.text.split` on it

if you want `|mode1=A|mode2=B|...` then `getAllArgsForBase` is certainly one of the better options to use

basically the implementation is pretty identical with one you use the split to get the table for the other you use the getAllArgsForBase function
after that you just loop through them